File: my-submissions/m50.java
class Solution {
    public double myPow(double x, int n) {
        return myPow(x, (long) n);

    }
    private double myPow(double x, long n) {
        if (n == 0) 
            return 1.;

        if (n < 0)
            return 1 / myPow(x, n * -1);
        
        if (n == 1)
            return x;

        double half = myPow(x, n / 2);
        long remainder = n % 2;

        if (remainder == 1)
            return half * half * x;
        return half * half;
    }
}

File: my-submissions/m50.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if not n :
            return 1
        
        if n == 1 :
            return x
        
        if n < 0 :
            return 1 / self.myPow(x, -n)

        half = n // 2
        remainder = n - half * 2
        half = self.myPow(x, half)

        if remainder :
            return x * half * half
        
        return half * half